File: app/src/main/java/com/sungkyul/osan/MainActivity.java
package com.sungkyul.osan;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    ImageView baldal,jungsin,service,carservice,association,mapimg,welfare,job,etc;
    TextView baldaltext,jungsintext,servicetext,carservicetext,associationtext,maptext,welfaretext,jobtext,etctext,allfacil;
    TextView copyrights,sosik;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        copyrights = findViewById(R.id.info);

        copyrights.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), Copyrights.class);
                startActivity(intent);
            }
        });

        allfacil = findViewById(R.id.allfacil);

        allfacil.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category","모든 시설 조회");
                startActivity(intent);
            }
        });

        baldal = findViewById(R.id.baldal);
        baldaltext = findViewById(R.id.baldaltext);

        baldal.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",baldaltext.getText().toString());
                startActivity(intent);
            }
        });
        baldaltext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",baldaltext.getText().toString());
                startActivity(intent);
            }
        });

        jungsin = findViewById(R.id.jungsin);
        jungsintext = findViewById(R.id.jungsintext);

        jungsin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",jungsintext.getText().toString());
                startActivity(intent);
            }
        });
        jungsintext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",jungsintext.getText().toString());
                startActivity(intent);
            }
        });

        service = findViewById(R.id.service);
        servicetext = findViewById(R.id.servicetext);

        service.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",servicetext.getText().toString());
                startActivity(intent);
            }
        });
        servicetext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",servicetext.getText().toString());
                startActivity(intent);
            }
        });

        carservice = findViewById(R.id.carservice);
        carservicetext = findViewById(R.id.carservicetext);

        carservice.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CarService.class);
                intent.putExtra("category",carservicetext.getText().toString());
                startActivity(intent);
            }
        });
        carservicetext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CarService.class);
                intent.putExtra("category",carservicetext.getText().toString());
                startActivity(intent);
            }
        });

        association = findViewById(R.id.association);
        associationtext = findViewById(R.id.associationtext);

        association.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",associationtext.getText().toString());
                startActivity(intent);
            }
        });
        associationtext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",associationtext.getText().toString());
                startActivity(intent);
            }
        });


        mapimg = findViewById(R.id.mapimg); //map 이미지버튼 아이디
        maptext = findViewById(R.id.maptext);

        mapimg.setOnClickListener(new View.OnClickListener() { //map 이미지 선택시 map 인텐트
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), com.sungkyul.osan.map.Kakaomap.class);
                startActivity(intent);
            }
        });
        maptext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), com.sungkyul.osan.map.Kakaomap.class);
                startActivity(intent);
            }
        });

        welfare = findViewById(R.id.welfare);
        welfaretext = findViewById(R.id.welfaretext);

        welfare.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",welfaretext.getText().toString());
                startActivity(intent);
            }
        });
        welfaretext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",welfaretext.getText().toString());
                startActivity(intent);
            }
        });

        job = findViewById(R.id.job);
        jobtext = findViewById(R.id.jobtext);

        job.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",jobtext.getText().toString());
                startActivity(intent);
            }
        });
        jobtext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",jobtext.getText().toString());
                startActivity(intent);
            }
        });

        etc = findViewById(R.id.etc);
        etctext = findViewById(R.id.etctext);

        etc.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",etctext.getText().toString());
                startActivity(intent);
            }
        });
        etctext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CategoryClick.class);
                intent.putExtra("category",etctext.getText().toString());
                startActivity(intent);
            }
        });

        sosik = findViewById(R.id.sosik);
        sosik.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), SosikActivity.class);
                intent.putExtra("title","오산시 소식");
                startActivity(intent);
            }
        });
    }
}

File: KotlinAndroidStudies/CardView_Intent_Activity/app/src/main/java/com/youngsun/cardview_intent_activity/MainActivity.kt
package com.youngsun.cardview_intent_activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 이미지를 클릭했을 때, 이미지마다 각각 Activity를 할당시켜주지 않고,
        // 하나의 Activity인 ImageActivity로 클릭된 Image의 데이터를 전달(Intent)하여 이미지를 크게 보여준다.

        val member1 = findViewById<ImageView>(R.id.member1)
        member1.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",1)
            startActivity(intent)

        }

        val member2 = findViewById<ImageView>(R.id.member2)
        member2.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",2)
            startActivity(intent)

        }

        val member3 = findViewById<ImageView>(R.id.member3)
        member3.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",3)
            startActivity(intent)

        }

        val member4 = findViewById<ImageView>(R.id.member4)
        member4.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",4)
            startActivity(intent)

        }

        val member5 = findViewById<ImageView>(R.id.member5)
        member5.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",5)
            startActivity(intent)

        }

        val member6 = findViewById<ImageView>(R.id.member6)
        member6.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",6)
            startActivity(intent)

        }

        val member7 = findViewById<ImageView>(R.id.member7)
        member7.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",7)
            startActivity(intent)

        }

        val member8 = findViewById<ImageView>(R.id.member8)
        member8.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",8)
            startActivity(intent)

        }

        val member9 = findViewById<ImageView>(R.id.member9)
        member9.setOnClickListener {
            val intent = Intent( this, ImageActivity::class.java)
            intent.putExtra("data",9)
            startActivity(intent)

        }
    }
}